public/js/components/Rep.js

```diff
@@ -0,0 +1,13 @@
+"use strict";
+
+
+// Node does not support the AMD syntax used in Rep
+// Node also does not support Worker, which is an issue when I built the Rep with webpack
+if (typeof window == "object") {
+    const Rep = require("../lib/devtools/client/shared/components/reps/Rep").Rep;
+    require("../lib/devtools/client/shared/components/reps/reps.css");
+
+    module.exports = Rep;
+} else {
+    module.exports = function() {};
+}
```

Comment:
> add a typeof check for Worker and then build Rep.

Hmm, I don't have a better tip, although could you somehow define the `define` function in the global scope?

> The ideal solution would be to convert Rep to commonjs, but I'm not sure if that would be possible...

The problem is that Reps are consumed by the JSON viewer. JSON Viewer runs in a page and needs a way to load all the modules (async). As soon as we allow using WebPack I can build jsonviewer.bundle.js (just like you do for the debugger.html), include it in JSON viewer page and we can remove the AMD support.

Honza